jerry-core/vm/vm.c

```diff
@@ -847,6 +847,31 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
 
         if (opcode_data & VM_OC_BACKWARD_BRANCH)
         {
+#ifdef JERRY_VM_EXEC_STOP
+          if (JERRY_CONTEXT (vm_exec_stop_cb) != NULL)
+          {
+            if (--JERRY_CONTEXT (vm_exec_stop_counter) == 0)
```

Comment:
It is completely well-defined and standard-specified, there is no vagueness involved. `&&` in C has short circuit evaluation. Please see ISO/IEC 9899 standard, section 6.5.13, "Logical AND operator" (http://www.open-std.org/JTC1/SC22/WG14/www/docs/n1256.pdf, http://www.open-std.org/JTC1/SC22/WG14/www/standards.html#9899)